### PR TITLE
[Keyboard] Add  yandrstudio/transition80

### PR DESCRIPTION
**Pull request by jiaxin96**
## Description

Add  yandrstudio/transition80 keyboard.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yandrstudio/transition80/transition80.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    LOCK_GUI = SAFE_RANGE,
+#endif
+    TOG_MACOS_KEYMAP,
+    KC_MISSION_CONTROL,
+    KC_LAUNCHPAD,
+#ifdef VIA_ENABLE
+    NEW_SAFE_RANGE = SAFE_RANGE  // Important!
+#else
+    NEW_SAFE_RANGE   // Important!
+#endif
```
Comment (drashna) [inline keyboards/yandrstudio/transition80/transition80.h:37]: Not needed.  User keycodes has a separate block now, starting with `QK_USER`. 
```suggestion
```

Also, 19697 changes how some of this is handled (eg, much cleaner and more in line with QMK, in general).

File: keyboards/yandrstudio/transition80/transition80.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        case KC_LG:
+            if (record->event.pressed) {
+                process_magic(GUI_TOG, record);
+            }
+            return false;
+        case KC_MACOS:
+            if (record->event.pressed) {
+                process_magic(CG_TOGG, record);
+            }
+            return false;
```
Comment (zvecr) [inline keyboards/yandrstudio/transition80/transition80.c:18]: Why is this reimplementing core functionality?

Comment (jiaxin96) [reply keyboards/yandrstudio/transition80/transition80.c]: In order to display these keys in VIA, and the definitions of these keys have been put into via/keymap.c now.

Comment (zvecr) [reply keyboards/yandrstudio/transition80/transition80.c]: These keycode have static values, and this code is unnecessary and should be removed. If this needs to be fixed in via, then thats not a problem for this repo, and should be fixed there instead.

Comment (jiaxin96) [reply keyboards/yandrstudio/transition80/transition80.c]: Well, it is really difficult to wait or push the update of via. Anyway, I have deleted this code according to your suggestion.

**Review by drashna — APPROVED** [review]

File: keyboards/yandrstudio/transition80/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                host_consumer_send(0x2A0);
+            } else {
+                host_consumer_send(0);
+            }
+            return false;
+        default:
+            break;
+    }
+    return true;
+}
```
Comment (zvecr) [inline keyboards/yandrstudio/transition80/keymaps/via/keymap.c:69]:
```suggestion
```

File: keyboards/yandrstudio/transition80/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    LOCK_GUI = QK_USER,
+    TOG_MACOS_KEYMAP,
+    KC_MISSION_CONTROL,
+    KC_LAUNCHPAD
+};
+
+#define KC_LG    LOCK_GUI
+#define KC_MACOS TOG_MACOS_KEYMAP
+#define KC_MCTL  KC_MISSION_CONTROL
+#define KC_LPAD  KC_LAUNCHPAD
```
Comment (zvecr) [inline keyboards/yandrstudio/transition80/keymaps/via/keymap.c:16]:
```suggestion
```

File: keyboards/yandrstudio/transition80/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        KC_CAPS, KC_A,    KC_S,    KC_D,   KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,   KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM,KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                           KC_SPC,                            KC_RALT, MO(1),   KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RIGHT),
+    [1] = LAYOUT(
+        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MACOS,KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,
```
Comment (zvecr) [inline keyboards/yandrstudio/transition80/keymaps/via/keymap.c:31]:
```suggestion
        KC_TRNS,          KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/yandrstudio/transition80/keymaps/via/keymap.c (hunk 5 above)
Comment (fauxpark) [reply keyboards/yandrstudio/transition80/keymaps/via/keymap.c:31]: Can be `AG_TOGG` instead.

File: keyboards/yandrstudio/transition80/transition80.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2022 Y&R-Biu (@jiaxin96)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/yandrstudio/transition80/transition80.h:7]: This can be moved to info.json, then these two files can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**